linera-execution/src/wasm/mod.rs

```diff
@@ -27,33 +27,28 @@ use self::{runtime_actor::RuntimeActor, sanitizer::sanitize};
 use crate::{
     ApplicationCallResult, Bytecode, CalleeContext, ContractRuntime, ExecutionError,
     MessageContext, OperationContext, QueryContext, RawExecutionResult, ServiceRuntime,
-    SessionCallResult, SessionId, UserApplication, WasmRuntime,
+    SessionCallResult, SessionId, UserContract, UserService, WasmRuntime,
 };
 use async_lock::RwLock;
 use async_trait::async_trait;
 use futures::future;
 use std::{path::Path, sync::Arc};
 use thiserror::Error;
 
-/// A user application in a compiled WebAssembly module.
-pub enum WasmApplication {
+/// A user contract in a compiled WebAssembly module.
+pub enum WasmContract {
```

Comment:
Nit: Maybe the variants don't need named fields anymore? It could just be `Wasmer(::wasmer::Enginer, ::wasmer::Module)` and `Wasmtime(Arc<::wasmtime::Module>)`.

Reply:
sure we can do that

Reply:
As a matter of fact, I prefer to have named fields since we have `engine` and `module` in one case.